index.js

```diff
@@ -1,30 +1,50 @@
 'use strict';
 
+class PriorityQueue {
+	constructor() {
+		this._queue = [];
+	}
+	put(run, priority) {
+		priority = priority || 0;
+		this._queue.push({priority, run});
+		this._queue.sort((a, b) => {
+			return b.priority - a.priority;
+		});
```

Comment:
@sindresorhus we can improve run time of this with `lower_bound`:

```js
function lower_bound(arr, value, comp) {
	let first = 0;
	let last = arr.length;
	let count = last - first;
	while (count > 0) {
		let step = (count / 2) | 0;
		let it = first + step;
		if (comp(arr[it], value)) {
			first = ++it;
			count -= step + 1;
		} else {
			count = step;
		}
	}
	return first;
}
```

and rewrite `put` method:

```js
		const element = {priority, run};
		const index = lower_bound(this._queue, 0, element, (a, b) => b.priority - a.priority);
		this._queue.splice(index, 0, element);
```

Should we go for it?